shapely/_geometry.py

```diff
@@ -731,20 +731,23 @@ def set_precision(geometry, grid_size, mode="valid_output", **kwargs):
 
     By default, geometries use double precision coordinates (grid_size = 0).
 
-    Coordinates will be rounded if a precision grid is less precise than the
-    input geometry. Duplicated vertices will be dropped from lines and
-    polygons for grid sizes greater than 0. Line and polygon geometries may
-    collapse to empty geometries if all vertices are closer together than
-    grid_size. Z values, if present, will not be modified.
-
-    Note: subsequent operations will always be performed in the precision of
-    the geometry with higher precision (smaller "grid_size"). That same
-    precision will be attached to the operation outputs.
-
-    Also note: input geometries should be geometrically valid; unexpected
-    results may occur if input geometries are not.
-
-    Returns None if geometry is None.
+    Coordinates will be rounded if the precision grid specified is less precise than the
+    input geometry. Duplicated vertices will be dropped from lines and polygons for grid
+    sizes greater than 0. Line and polygon geometries may collapse to empty geometries
+    if all vertices are closer together than ``grid_size`` or if a polygon becomes
+    significantly narrower than ``grid_size``. Spikes or sections in polygons narrower
+    than ``grid_size`` after rounding the vertices will be removed, which can lead to
+    multipolygons or empty geometries. Z values, if present, will not be modified.
+
+    Notes:
+
+    * subsequent operations will always be performed in the precision of
+      the geometry with higher precision (smaller "grid_size"). That same
+      precision will be attached to the operation outputs.
+    * input geometries should be geometrically valid; unexpected results may
+      occur if input geometries are not.
+    * the geometry returned will be in :ref:`mild canonical form <canonical-form>`.
```

Comment:
```suggestion
    * the geometry returned will be in :ref:`mild canonical form <canonical-form>`,
      and the order of vertices can change and should not be relied upon.
```

(I think that's the main gotcha that triggered some issue? So in addition to putting the link, summarizing it like that might be helpful)